unit-tests-java8/src/test/java/org/eclipse/collections/test/stack/StackIterableTestCase.java

```diff
@@ -85,7 +86,7 @@ default void InternalIterable_forEachWith()
     @Test
     default void RichIterable_getFirst_and_getLast()
     {
-        assertThrows(UnsupportedOperationException.class, this.newWith()::getLast);
+        assertThrows(EmptyStackException.class, this.newWith()::getLast);
     }
 
```

Comment:
This override is no longer necessary. We can just delete the whole method RichIterable_getFirst_and_getLast here. Afterwards, we'll inherit the assertions for the superclass:

```java
    @Test
    default void RichIterable_getFirst_and_getLast()
    {
        RichIterable<Integer> iterable = this.newWith(3, 2, 1);
        assertNotEquals(iterable.getFirst(), iterable.getLast());
    }
```

Reply:
`RichIterable` & `OrderedIterable` expect null when an element doesn't exists but in Stack we have EmptyStack exception.

Reply:
If we were to return null on getLast(), we need to return null on getFirst() too.